myTrip/tests/unittests/registration/test_views.py

```diff
@@ -0,0 +1,134 @@
+from datetime import datetime
+from json import dumps
+from django.test import TestCase
+from django.urls import reverse
+
+from registration.models import CustomUser
+
+
+class RegistrationViewsTests(TestCase):
+    """
+    Test 'login', 'logout' and 'register' views.
+    """
+
+    def setUp(self):
+        """
+        Create CustomUser record in database.
+        """
+
+        date = datetime.now()
+        self.user = CustomUser.objects.create(id=5,
+                                              email='test@gmail.com',
+                                              created = date,
+                                              last_modified = date)
+        self.user.set_password('password')
+        self.user.save()
+
+    def test_login_bad_request(self):
+        """
+        Test 'login' view with bad request method.
+        """
+
+        request = self.client.get(reverse('login_view'))
+        self.assertEqual(request.status_code, 400)
+
+    def test_login_success(self):
+        """
+        Test 'login' view.
+        """
+
+        request = self.client.post(reverse('login_view'),
+                                   dumps({"email": "test@gmail.com", "password": "password"}),
+                                   content_type='application/json')
+        self.assertEqual(request.status_code, 200)
+
+    def test_login_wrong_email(self):
+        """
+        Test 'login' view with wrong email.
+        """
+
+        request = self.client.post(reverse('login_view'),
+                                   dumps({"email": "krest@gmail.com", "password": "password"}),
+                                   content_type='application/json')
+        self.assertEqual(request.status_code, 403)
+
+    def test_login_wrong_password(self):
+        """
+        Test 'login' view view with wrong password.
+        """
+
+        request = self.client.post(reverse('login_view'),
+                                   dumps({"email": "test@gmail.com", "password": "pwd"}),
+                                   content_type='application/json')
+        self.assertEqual(request.status_code, 403)
+
+    def test_login_wrong_email_and_password(self):
+        """
+        Test 'login' view with wrong email & password.
+        """
+
+        request = self.client.post(reverse('login_view'),
+                                   dumps({"email": "krest@gmail.com", "password": "pwd"}),
+                                   content_type='application/json')
+        self.assertEqual(request.status_code, 403)
+
+    def test_logout_bad_request(self):
+        """
+        Test 'logout' view with bad request method.
+        """
+
+        request = self.client.post(reverse('logout_view'))
+        self.assertEqual(request.status_code, 400)
+
+    def test_logout_success(self):
+        """
+        Test 'logout' view.
+        """
+
+        self.client.login(username='test@gmail.com', password='password')
+
+        request = self.client.get(reverse('logout_view'))
```

Comment:
@black-art reverse is used to access a address by "name" attribute instead of full url.
Here we use logout_view to actually log the user out :)